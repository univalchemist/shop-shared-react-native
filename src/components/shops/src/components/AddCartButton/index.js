import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import { Box, Footer, TrackedButton } from '@shops/wrappers/components';
import { connect } from 'react-redux';
import { Spinner } from '@shops/components';
import { useIntl, useTheme } from '@shops/wrappers/core/hooks';
import { addToCart } from '@shops/store/actions';
import PropTypes from 'prop-types';

export const AddCartButton = ({
  addToCart,
  quantity,
  sku,
  onAddToCartSuccess = () => {},
  onAddToCartFailed = () => {},
}) => {
  const intl = useIntl();
  const theme = useTheme();
  const [isLoading, setIsLoading] = useState(false);
  const onPressAddToCart = async () => {
    try {
      setIsLoading(true);
      await addToCart({ sku, quantity });
      onAddToCartSuccess();
    } catch (e) {
      onAddToCartFailed(e);
    } finally {
      setIsLoading(false);
    }
  };
  if (isLoading) return <Spinner />;
  return (
    <Footer
      flexDirection="row"
      style={[styles.footer, { backgroundColor: theme.colors.white }]}
    >
      <Box flex={1} flexDirection="row">
        <Box flex={1}>
          <TrackedButton
            primary
            onPress={onPressAddToCart}
            title={intl.formatMessage({
              id: 'shop.productDetail.addToCart',
              defaultMessage: 'Add to Cart',
            })}
          />
        </Box>
      </Box>
    </Footer>
  );
};

const styles = StyleSheet.create({
  footer: {
    paddingBottom: 16,
  },
});

AddCartButton.propTypes = {
  sku: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
  onAddToCartSuccess: PropTypes.func,
  onAddToCartFailed: PropTypes.func,
};

export default connect(null, { addToCart })(AddCartButton);
