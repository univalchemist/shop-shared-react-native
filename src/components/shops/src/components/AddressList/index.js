export { default as BillingAddresses } from './BillingAddresses';
export { default as ShippingAddresses } from './ShippingAddresses';
export { default as AddressForm } from './AddressForm';
