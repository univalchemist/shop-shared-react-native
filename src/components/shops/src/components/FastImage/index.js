import React from 'react';
import {
  flex,
  justifySelf,
  alignSelf,
  order,
  display,
  maxWidth,
  minWidth,
  width,
  height,
  maxHeight,
  minHeight,
  position,
  size,
  space,
  resizeMode,
  accessibilityLabel,
} from 'styled-system';
import styled from 'styled-components/native';
import RNFastImage from 'react-native-fast-image';

const BaseImage = styled(RNFastImage)`
  ${space}
  ${flex}
  ${justifySelf}
  ${alignSelf}
  ${display}
  ${width}
  ${maxWidth}
  ${minWidth}
  ${position}
  ${height}
  ${maxHeight}
  ${minHeight}
  ${size}
  ${order}
  ${resizeMode}
  ${accessibilityLabel}
`;

const FastImage = React.forwardRef(
  ({ secure, authToken, source, ...props }, ref) => {
    const headers = { Authorization: `Bearer ${authToken}` };

    const imageSource = secure
      ? {
          ...source,
          headers,
        }
      : source;

    return <BaseImage ref={ref} source={imageSource} {...props} />;
  },
);

export default FastImage;
