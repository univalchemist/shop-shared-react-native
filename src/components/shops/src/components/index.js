export { RadioButton, RadioGroup } from './Radio';
export { default as ShopHeader } from './Header';
export { default as Spinner } from './Spinner';
export { default as AddToCartButton } from './AddCartButton';
export { default as BackButton } from './BackButton';
export { default as ImageProduct } from './ImageProduct';
export { default as RatingStars } from './RatingStars';
export { default as Price } from './Price';
export { default as ProductItem } from './ProductItem';
export { default as ClearAllButton } from './ClearAllButton';
export { default as BorderInput } from './BorderInput';
export { default as SnackBar } from './SnackBar';
export { default as ProductHeader } from './ProductHeader';
export { default as QuantityButton } from './QuantityButton';
export {
  BillingAddresses,
  ShippingAddresses,
  AddressForm,
} from './AddressList';
export { default as CheckBox } from './CheckBox';
export { default as SearchInput } from './SearchInput';
