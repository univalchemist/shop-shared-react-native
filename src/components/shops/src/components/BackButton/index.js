import React from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { Box, Text, Icon } from '@shops/wrappers/components';
import { useIntl } from '@shops/wrappers/core/hooks';

export const BackButton = ({ navigation, labelId = 'shop.common.back' }) => {
  const intl = useIntl();
  return (
    <Box px={12} py={12}>
      <TouchableOpacity
        style={styles.buttonContainer}
        onPress={() => navigation?.goBack()}
      >
        <Box flexDirection="row">
          <Icon name="keyboard-arrow-left" />
          <Text fontSize={16}>
            {intl.formatMessage({
              id: labelId,
              defaultMessage: 'Back',
            })}
          </Text>
        </Box>
      </TouchableOpacity>
    </Box>
  );
};

const styles = StyleSheet.create({
  buttonContainer: { width: 50 },
});

export default BackButton;
