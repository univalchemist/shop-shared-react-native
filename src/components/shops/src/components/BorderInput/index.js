import React from 'react';
import { Box, PlainText } from '@shops/wrappers/components';
import styled from 'styled-components/native';
import { useTheme } from '@shops/wrappers/core/hooks';
import { TextInput } from 'react-native';
import {
  fontFamily,
  textAlign,
  lineHeight,
  fontWeight,
  fontSize,
  letterSpacing,
  color,
  space,
  fontStyle,
  marginTop,
  paddingTop,
  paddingBottom,
  flexWrap,
  width,
  marginRight,
  marginLeft,
  borderRadius,
  borderWidth,
  borderBottom,
  borderColor,
} from 'styled-system';

const StyledInput = styled(TextInput)`
    ${fontSize}
    ${fontFamily}
    ${textAlign}
    ${lineHeight}
    ${fontWeight}
    ${letterSpacing}
    ${color}
    ${space}
    ${fontStyle}
    ${marginRight}
    ${marginLeft}
    ${marginTop}
    ${paddingTop}
    ${paddingBottom}
    ${flexWrap}
    ${width}
    ${borderRadius}
    ${borderWidth}
    ${borderBottom}
    ${borderColor}
`;

export const BorderInput = Props => {
  const {
    onChangeText,
    placeholder,
    borderColor,
    value,
    optional,
    outlined,
    label,
    isError,
    color,
  } = Props;
  const theme = useTheme();
  return (
    <Box alignSelf={'center'}>
      <StyledInput
        {...Props}
        py={12}
        px={10}
        color={color}
        onChangeText={onChangeText}
        placeholder={placeholder}
        borderWidth={isError ? 2 : 1}
        value={value + ''}
        borderRadius={5}
        borderColor={isError ? theme.colors.primary[0] : theme.colors.border}
      />
      {outlined && (
        <Box
          alignItems={'center'}
          left={12}
          px={2}
          backgroundColor={theme.colors.white}
          position={'absolute'}
        >
          <PlainText color={isError && theme.colors.red} fontSize={12}>
            {label}
          </PlainText>
        </Box>
      )}
      {optional && (
        <PlainText fontSize={12} marginLeft={12}>
          {'Optional'}
        </PlainText>
      )}
    </Box>
  );
};

export default BorderInput;
