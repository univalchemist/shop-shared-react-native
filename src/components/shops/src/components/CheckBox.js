import React from 'react';
import { StyleSheet, TouchableWithoutFeedback } from 'react-native';
import { CheckBox } from 'react-native-elements';
import { SecondaryText, Image, Flex } from '@shops/wrappers/components';
import { checkboxActive, checkboxInactive } from '@shops/assets/icons';

const CustomCheckBox = ({ onPress, label, checked }) => {
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <Flex flexDirection="row" justifyContent="space-between">
        <SecondaryText>{label}</SecondaryText>
        <CheckBox
          containerStyle={styles.checkBoxContainer}
          checked={checked}
          onPress={onPress}
          checkedIcon={<Image source={checkboxActive} />}
          uncheckedIcon={<Image source={checkboxInactive} />}
        />
      </Flex>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  checkBoxContainer: {
    padding: 0,
    marginTop: 0,
    marginBottom: 0,
    marginLeft: 0,
    marginRight: 0,
  },
});

export default CustomCheckBox;
