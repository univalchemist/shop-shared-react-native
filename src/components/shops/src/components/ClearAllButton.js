import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Text } from '@shops/wrappers/components';

const ClearAllButton = ({ onPress, text, navigation }) => {
  const overridedOnPress = () => {
    onPress && onPress();
    navigation.goBack();
  };

  return (
    <TouchableOpacity onPress={overridedOnPress}>
      <Text mr={16}>{text}</Text>
    </TouchableOpacity>
  );
};

export default ClearAllButton;
