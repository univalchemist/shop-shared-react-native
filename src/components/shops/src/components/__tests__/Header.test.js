import React from 'react';
import { renderForTest } from '@testUtils';
import { ShopHeader } from '../index';
import { act, fireEvent } from 'react-native-testing-library';
import { SHOP_CART_SCREEN } from '@shops/navigation/routes';
import ModalDropdown from 'react-native-modal-dropdown';
import { Image } from '@shops/wrappers/components';
import { menu as MenuImg, xmark as xMarkIcon } from '@shops/assets/icons';
import { Platform } from 'react-native';
import * as utils from '@utils/isIphoneX';
import navigation from '@testUtils/__mocks__/navigation';

const initialState = {
  shop: {
    cart: {},
  },
};

describe('Header', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation(isSearch => [isSearch, setState]);

  test('should match snapshot', async () => {
    const navigation = {
      navigate: jest.fn(),
    };
    const Component = renderForTest(<ShopHeader navigation={navigation} />, {
      initialState,
    });
    expect(Component.toJSON()).toMatchSnapshot();
  });

  test('should show menu properly', async () => {
    const navigation = {
      navigate: jest.fn(),
    };
    const { getByTestId, getByType, getAllByType } = renderForTest(
      <ShopHeader navigation={navigation} />,
      {
        initialState,
      },
    );
    const modalDropdown = getByType(ModalDropdown);
    act(() => {
      fireEvent(modalDropdown, 'dropdownWillShow');
    });
    const xMarkmage = getByType(Image);
    expect(xMarkmage.props.source).toBe(xMarkIcon);
    act(() => {
      fireEvent(modalDropdown, 'dropdownWillHide');
    });
    const menuImage = getAllByType(Image)[0];
    expect(menuImage.props.source).toBe(MenuImg);
  });

  test('should navigate when cart icon is pressed', async () => {
    const navigation = {
      navigate: jest.fn(),
    };
    const { getByTestId } = renderForTest(
      <ShopHeader navigation={navigation} />,
      {
        initialState,
      },
    );
    const HeaderButton = getByTestId('cartIcon');
    act(() => {
      fireEvent.press(HeaderButton);
    });

    expect(navigation.navigate).toBeCalled();
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledWith(SHOP_CART_SCREEN);
  });

  test('should have height 56 when cart platform is android', async () => {
    const navigation = {
      navigate: jest.fn(),
    };
    Platform.OS = 'android';
    const { getByTestId } = renderForTest(
      <ShopHeader navigation={navigation} />,
      {
        initialState,
      },
    );
    const ContainerHeader = getByTestId('boxContainer');
    expect(ContainerHeader.props.height).toBe(56);
  });

  test('should have height 88 when device is iPhoneX', async () => {
    Platform.OS = 'ios';
    jest.spyOn(utils, 'isIphoneX').mockReturnValue(true);
    const { getByTestId } = renderForTest(<ShopHeader />, {
      initialState,
    });
    const ContainerHeader = getByTestId('boxContainer');
    expect(ContainerHeader.props.height).toBe(88);
  });

  test('should navigate correctly when user press', async () => {
    const { getByType } = renderForTest(
      <ShopHeader navigation={navigation} />,
      {
        initialState,
      },
    );
    const ModalDropdownComp = getByType(ModalDropdown);
    fireEvent(ModalDropdownComp, 'renderSeparator');
    fireEvent(ModalDropdownComp, 'select', 1);
    fireEvent(ModalDropdownComp, 'renderRow', {
      image: 'imageUrl',
      titleId: 'shop.menu.shopHome',
    });

    expect(navigation.navigate).toBeCalledWith('ORDER_HISTORY_SCREEN');
  });


  // test('should not navigate method when search icon is pressed', async () => {
  //   const navigation = {
  //     navigate: jest.fn(),
  //   };
  //   const { getByTestId } = renderForTest(
  //     <ShopHeader navigation={navigation} />, {
  //       initialState,
  //     }
  //   );
  //   const HeaderButton = getByTestId('searchIcon');
  //   const boxContainer = getByTestId('boxContainer');
  //   act(() => {
  //     fireEvent.press(HeaderButton);
  //   });
  //   expect(boxContainer.props.backgroundColor).toEqual(theme.colors.gray[0]);
  //   expect(navigation.navigate).not.toBeCalled();
  //   expect(navigation.navigate).toHaveBeenCalledTimes(0);
  // });

  // test('should navigate method when onSubmitEditing is called ', async () => {
  //   const navigation = {
  //     navigate: jest.fn(),
  //   };
  //   const { getByTestId } = renderForTest(
  //     <ShopHeader navigation={navigation} />, {
  //       initialState,
  //     }
  //   );
  //
  //   const HeaderButton = getByTestId('searchIcon');
  //   act(() => HeaderButton.props.onPress());
  //
  //   const SearchBox = getByTestId('searchBox');
  //   act(() => {
  //     SearchBox.props.onSubmitEditing();
  //     fireEvent(SearchBox, 'onChangeText', 'test');
  //   });
  //   expect(navigation.navigate).toBeCalled();
  //   expect(navigation.navigate).toHaveBeenCalledWith(
  //     NAVIGATION_SEARCH_SCREEN_PATH,
  //   );
  // });
});
