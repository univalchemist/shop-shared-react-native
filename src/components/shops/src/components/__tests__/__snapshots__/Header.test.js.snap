// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        height={64}
                        style={
                          Array [
                            Object {
                              "height": 64,
                            },
                          ]
                        }
                        testID="boxContainer"
                      >
                        <View
                          alignItems="center"
                          backgroundColor="#212121"
                          flex={1}
                          flexDirection="row"
                          justifyContent="space-between"
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#212121",
                                "flexBasis": 0,
                                "flexDirection": "row",
                                "flexGrow": 1,
                                "flexShrink": 1,
                                "justifyContent": "space-between",
                              },
                            ]
                          }
                        >
                          <View
                            animated={true}
                            defaultIndex={-1}
                            defaultValue="Please select..."
                            disabled={false}
                            dropdownStyle={
                              Object {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                                "height": 350,
                                "marginTop": 16,
                                "width": "100%",
                              }
                            }
                            keyboardShouldPersistTaps="never"
                            onDropdownWillHide={[Function]}
                            onDropdownWillShow={[Function]}
                            onSelect={[Function]}
                            options={
                              Array [
                                Object {
                                  "image": Object {
                                    "testUri": "../../../src/components/shops/src/assets/icons/shop.png",
                                  },
                                  "navigateTo": "SHOP_HomeScreen",
                                  "titleId": "shop.menu.shopHome",
                                },
                                Object {
                                  "image": Object {
                                    "testUri": "../../../src/components/shops/src/assets/icons/order.png",
                                  },
                                  "navigateTo": "ORDER_HISTORY_SCREEN",
                                  "titleId": "shop.menu.orderHistory",
                                },
                                Object {
                                  "image": Object {
                                    "testUri": "../../../src/components/shops/src/assets/icons/credit.png",
                                  },
                                  "navigateTo": "SHOP_HomeScreen",
                                  "titleId": "shop.menu.creditCards",
                                },
                                Object {
                                  "image": Object {
                                    "testUri": "../../../src/components/shops/src/assets/icons/delivery.png",
                                  },
                                  "navigateTo": "SHOP_DELIVERY_ADDRESSES_SCREEN",
                                  "titleId": "shop.menu.deliveryAddress",
                                },
                              ]
                            }
                            renderRow={[Function]}
                            renderSeparator={[Function]}
                            scrollEnabled={true}
                            showsVerticalScrollIndicator={true}
                            testID="modalDropdown"
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              isTVSelectable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <Image
                                height={12}
                                ml={16}
                                resizeMode="contain"
                                source={
                                  Object {
                                    "testUri": "../../../src/components/shops/src/assets/icons/menu.png",
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "height": 12,
                                      "marginLeft": 16,
                                      "width": 18,
                                    },
                                  ]
                                }
                                width={18}
                              />
                            </View>
                          </View>
                          <View
                            alignItems="center"
                            flexDirection="row"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                },
                              ]
                            }
                          >
                            <View
                              mr={16}
                              style={
                                Array [
                                  Object {
                                    "marginRight": 16,
                                  },
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {},
                                  ]
                                }
                                testID="cartIcon"
                              >
                                <View
                                  px={1}
                                  style={
                                    Array [
                                      Object {
                                        "paddingLeft": 4,
                                        "paddingRight": 4,
                                      },
                                    ]
                                  }
                                >
                                  <Image
                                    height={18}
                                    source={
                                      Object {
                                        "testUri": "../../../src/components/shops/src/assets/icons/cart.png",
                                      }
                                    }
                                    style={
                                      Array [
                                        Object {
                                          "height": 18,
                                          "width": 18,
                                        },
                                      ]
                                    }
                                    width={18}
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
